### ISSUE TITLE
JSDoc-style multi-line comments are wrongly indented

### ISSUE THREAD

**Issue by thennothinghappened** [microsoft/vscode-cpptools#12249]
### Environment

- OS and Version: MacOS Ventura 13.5
- VS Code Version: 1.88.1
- C/C++ Extension Version: v1.19.9
- If using SSH remote, specify OS of remote machine: N/A

### Bug Summary and Steps to Reproduce

**Bug Summary:** When writing a multi-line documentation comment, the ending line is not indented correctly.

**Steps to reproduce:**
1. On a new line, start a multi-line documentation comment: `/**`.
2. The closing `*/` will be autocompleted, placing the cursor right before it.
3. Press enter to expand the comment to a new line.
4. The closing `*/` will not be correctly indented with the contents of the comment:

```cpp
/**
 *
*/
```

**Expected behavior:**
I'd expect the closing comment line to be indented by one space to match with the comment body:

```cpp
/**
 *
 */
```

### Configuration and Logs

**Log Diagnostics:**

```
-------- Diagnostics - 24/04/2024, 08:42:08
Version: 1.19.9
Current Configuration:
{
    "name": "Mac",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "macFrameworkPath": [
        "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks"
    ],
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "macos-clang-arm64",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPath": "/usr/bin/clang++",
    "compilerPathIsExplicit": true,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
cpptools version (native): 1.19.9.0
No active translation units.

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 9124
```

**`c_cpp_properties.json`:**

N/A - Problem presents with no configuration given with no workspace needed.

**Language Server Log:**

```log
loggingLevel: Debug
loggingLevel has changed to: Debug
LSP: (invoked) cpptools/didChangeCppProperties (id: 6)
LSP: Sending response (id: 6)
LSP: (received) cpptools/didChangeVisibleTextEditors
LSP: (invoked) cpptools/didChangeVisibleTextEditors
LSP: (received) cpptools/didChangeVisibleTextEditors
LSP: (invoked) cpptools/didChangeVisibleTextEditors
LSP: (received) cpptools/didChangeVisibleTextEditors
LSP: (invoked) cpptools/didChangeVisibleTextEditors
LSP: (received) cpptools/didChangeVisibleTextEditors
LSP: (invoked) cpptools/didChangeVisibleTextEditors
```

### Other Extensions

Issue presents without any other extensions.

**Comment by sean-mcmanus** [comment]
@thennothinghappened I have a PR against VS Code at https://github.com/microsoft/vscode/pull/211202 which fixes this. You could also apply that PR manually to get the same behavior (although a VS Code update might overwrite those changes).

**Comment by sean-mcmanus** [comment]
@thennothinghappened This should be fixed in the latest VS Code Insiders release, although you may have to wait till tomorrow's build, and the fix to Doxygen generation with `/**` isn't available till 1.21.0.

**Comment by sean-mcmanus** [comment]
@thennothinghappened Fixed with the latest VS Code (and our doxygen comment generation with `/**` is fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/v1.21.0).
